Hw5

BlackRule opened the pull request.

BlackRule committed:
init

BlackRule committed:
GET http://localhost:3000/favicon.ico?favicon.0b3bf435.ico 500 (Internal Server Error)

BlackRule committed:
Server  Error: Route "/product/[id]" used `params.id`. `params` should be awaited before using its properties. Learn more: https://nextjs.org/docs/messages/sync-dynamic-apis
    at ProductPage (page.tsx:33:45)

BlackRule committed:
> Build error occurred
  [Error: Turbopack build failed with 2 errors:
  ./shared/features/about/About.module.scss.module.css
  Transforming CSS failed
  Selector "h1" is not pure. Pure selectors must contain at least one local class or id.

  ./shared/features/about/About.module.scss.module.css
  Transforming CSS failed
  Selector "p" is not pure. Pure selectors must contain at least one local class or id.
  ]

BlackRule committed:
Failed to compile.

  ./shared/components/MultiDropdown/MultiDropdown.tsx:66:19
  Type error: Argument of type 'RefObject<HTMLDivElement | null>' is not assignable to parameter of type 'RefObject<HTMLElement>'.
    Type 'HTMLDivElement | null' is not assignable to type 'HTMLElement'.
      Type 'null' is not assignable to type 'HTMLElement'.

    64 |
    65 |   const ref = useRef<HTMLDivElement>(null)
  > 66 |   useClickOutside(ref, () => setIsOpen(false))
       |                   ^
    67 |   useEffect( () => {
    68 |     // eslint-disable-next-line @typescript-eslint/no-unused-expressions
    69 |     disabled && setIsOpen(false)

BlackRule committed:
🔍  Inspect: https://vercel.com/blackrules-projects/next/623rRy7Ve1HfCED1aWLxWEkYyLPC [2s]
✅  Preview: https://next-ggz0lox4i-blackrules-projects.vercel.app [2s]

BlackRule committed:
Uncaught Error: Hydration failed because the server rendered text didn't match the client. As a result this tree will be regenerated on the client.
+ Find Now
- 🔍

BlackRule committed:
No images found! Can't build the thumb list without images. If you don't need thumbs, set showThumbs={false} in the Carousel. Note that it's not possible to get images rendered inside custom components. More info at https://github.com/leandrowd/react-responsive-carousel/blob/master/TROUBLESHOOTING.md

BlackRule committed:
Image with src "https://front-school.minio.ktsdev.ru/w3_Y8_Nw_Q_38c26c266e.jpeg" has either width or height modified, but not the other. If you use CSS to change the size of your image, also include the styles 'width: "auto"' or 'height: "auto"' to maintain the aspect ratio.
	(anonymous)	@	Product.tsx:45
Product	@	Product.tsx:43
<Product>
ProductPageContent	@	product-page-content.tsx:23
"use client"
ProductPage	@	page.tsx:53

BlackRule committed:
vercel

BlackRule committed:
A tree hydrated but some attributes of the server rendered HTML didn't match the client properties. This won't be patched up.
Added old repo link

BlackRule committed:
Revert "A tree hydrated but some attributes of the server rendered HTML didn't match the client properties. This won't be patched up."

This reverts commit d90c5b39d6affa9f8afba428ed6ba4a1d3102c9f.

BlackRule committed:
Added old repo link

sergromm commented on shared/components/Card/Card.module.scss:
```diff
@@ -0,0 +1,38 @@
+  }
+
+}
+.card__text{
```
нужно добавить `flex: 1`, тогда кнопки будут всегда прибиты к низу карточки

<img width="1003" height="384" alt="image" src="https://github.com/user-attachments/assets/1f2a3b5c-245a-4b17-8dd6-7a2d8762d72e" />

sergromm commented on shared/features/product/components/Product/Product.tsx:
```diff
@@ -0,0 +1,86 @@
+  return (
+    <section className={styles.product}>
+      {hasImages ? (
+        <Carousel
```
Тут параллакс какой-то случился, и сам слайдер слишком широкий стал

https://github.com/user-attachments/assets/7a5ec085-6e2e-4a46-a885-ad588e00144d

sergromm commented on shared/features/products/index.module.scss:
```diff
@@ -0,0 +1,58 @@
+  color: $text-accent;
+  font-weight: 700;
+}
+.productsGrid{
+  display: flex;
+  justify-content: center;
+  flex-wrap: wrap;
+  gap: 80px;
+  & a{
+    text-decoration: none;
+  }
+  @media (max-width: $bp-tablet) {
+    gap: 15px;
+  }
+}
```
я бы тут предложил на grid сделать и убрать у компонента с карточкой товара фиксированную ширину

```scss
.productsGrid {
  display: grid;
  gap: 1rem;
  
  // тут можно задать минимальную и максимальную ширину карточки
  grid-template-columns: repeat(3, minmax(200px, 1fr));

}
```

для адаптива можно выбрать сколько будет столбцов в сетеке 
```scss
  @media (max-width: $bp-tablet) {
    grid-template-columns: repeat(2, minmax(200px, 1fr));
  }
```

sergromm commented on shared/features/products/products-grid.tsx:
```diff
@@ -0,0 +1,44 @@
+          <Card
+            {...productToCardProps(product)}
+            actionSlot={
+              <Button
+                onClick={(event) => {
+                  event.preventDefault()
+                  event.stopPropagation()
+                  void cartStore.add(product)
+                }}
+              >
+                Add to Cart
+              </Button>
```
Всё ещё не хватает какого-то фидбека в ui, что товар добавлен в корзину. Можно например текст на время менять, если не хочется делать счётчик с количеством добавленных товаров

sergromm replied:
Так же относится к кнопкам на странице товара

sergromm commented on app/(components)/products-page-content.tsx:
```diff
@@ -0,0 +1,56 @@
+        name of the item
+      </Text>
+      <SearchBar />
+      <Filter className={styles.filter} options={filterOptions} />
```
Можно сломать ui фильтров, если отредактировать sc в url


https://github.com/user-attachments/assets/d1437d45-18a4-40f9-9203-48e93a5b7986

sergromm commented:
Не хватает отступов между фильтрами и поиском, и между пагинацией и сеткой товаров

<img width="568" height="154" alt="image" src="https://github.com/user-attachments/assets/c3ca12e5-8a7e-42da-a81a-e7690de734e0" />
<img width="301" height="587" alt="image" src="https://github.com/user-attachments/assets/d2d1eaf6-7881-4e20-9ccf-d93ff3644bed" />

sergromm commented:
К коду у меня нет комментов. Радует что настроил форматирование, благодарю. Оставил комменты в основном с багами, они не мешают дз принять, или предложениями

Дальше советую сфокусироваться на том чтобы навести красоту в проекте, над вёрсткой можно поработать. 

Так же подумать над UX корзины:

1) Сильно не хватает фидбека в ui, сейчас неочевидно добавился товар или нет 

2) Можно подумать над тем чтобы сохранять корзину когда пользователь логинится.
Сценарий такой:
- Я добавляю товары в корзину как анонимный пользователь
- Захожу на сайт
- Перехожу обратно в корзину и у меня пропадают все товары которые я добавлял до этого

Например можно запросить /add с тем id товаров которые уже хранил в localStorage

BlackRule committed:
A tree hydrated but some attributes of the server rendered HTML didn't match the client properties. This won't be patched up. This can happen if a SSR-ed Client Component used:
+href="/user"
-href="/login"
 <Link href={user ? '/user' : '/login'} className={styles.user} aria-label="Account" />

BlackRule committed:
IDK why but this is needed to properly size the carousel
